testnet/stacks-node/src/tests/epoch_23.rs

```diff
@@ -0,0 +1,626 @@
+use std::collections::HashMap;
+use std::env;
+use std::thread;
+
+use stacks::burnchains::Burnchain;
+use stacks::core::PEER_VERSION_EPOCH_2_2;
+use stacks::core::PEER_VERSION_EPOCH_2_3;
+use stacks::core::STACKS_EPOCH_MAX;
+use stacks::vm::types::QualifiedContractIdentifier;
+
+use crate::config::EventKeyType;
+use crate::config::EventObserverConfig;
+use crate::config::InitialBalance;
+use crate::neon;
+use crate::tests::bitcoin_regtest::BitcoinCoreController;
+use crate::tests::neon_integrations::*;
+use crate::tests::*;
+use crate::BitcoinRegtestController;
+use crate::BurnchainController;
+use stacks::core;
+
+use stacks::burnchains::PoxConstants;
+
+use clarity::vm::types::PrincipalData;
+
+#[test]
+#[ignore]
+/// Test the trait invocation behavior for contracts instantiated in epoch 2.05
+///  * in epoch 2.1: the trait invocation works
+///  * in epoch 2.2: trait invocation is broken, and returns a runtime error, even when wrapped
+///  * in epoch 2.3: the trait invocation works
+fn trait_invocation_behavior() {
```

Comment:
There isn't a simple way to test this in a unit test?  I'm pretty sure this is also an option (and it would be less flaky and run faster).

Reply:
The `clarity_vm::tests::contracts::trait_invocation_cross_epoch` unit test covers the behavior.

I think having both here is important -- because the integration test covers the miner path as well (also, using a real sortition db for the epoch switch).